### PR TITLE
Fixes #293

### LINKED ISSUE #293: Error while checking comments style within a string interpolation
If I write this line:
``` typescript
`${location.protocol}//${location.hostname}`
```
tslint writes: comment must start with a space
But it is a part of the new TypeScript 1.4 feature: the ES6 template strings, and not a `//` comment prefix.

### PR DESCRIPTION
**Pull request by gscshoyru**
Comment format rule should skip templates (and other things). And the
whitespace rule should skip templates too.

### CODE REVIEW

File: src/rules/commentFormatRule.ts [hunk 1]
```diff
@@ -27,8 +27,23 @@ export class Rule extends Lint.Rules.AbstractRule {
...
     public visitSourceFile(node: ts.SourceFile): void {
+        super.visitSourceFile(node);
         Lint.scanAllTokens(ts.createScanner(ts.ScriptTarget.ES5, false, node.text), (scanner: ts.Scanner) => {
+            var startPos = scanner.getStartPos();
+            if (this.tokensToSkipStartEndMap[startPos] != null) {
+                // tokens to skip are places where the scanner gets confused about what the token is, without the proper context
+                // (specifically, regex, identifiers, and templates). So skip those tokens.
+                scanner.setTextPos(this.tokensToSkipStartEndMap[startPos]);
+                return;
+            }
```
Comment (adidahiya) [inline src/rules/commentFormatRule.ts]: why not `else if`?

Comment (gscshoyru) [inline src/rules/commentFormatRule.ts]: There should probably be a newline between the two if blocks here -- but it shouldn't be an else if, the two things are different concepts (the first block is saying "if this is true, continue", which doesn't make sense to group with the next block.)

Comment (adidahiya) [inline src/rules/commentFormatRule.ts]: ok, fair :+1:

Comment (ashwinr) [inline src/rules/commentFormatRule.ts]: Yes new line please

File: src/rules/commentFormatRule.ts [hunk 2]
```diff
@@ -47,7 +62,30 @@ class CommentWalker extends Lint.RuleWalker {
...
+
+    public visitIdentifier(node: ts.Identifier) {
+        if (node.getStart() < node.getEnd()) {
+            // only add to the map nodes whose end comes after their start, to prevent infinite loops
+            this.tokensToSkipStartEndMap[node.getStart()] = node.getEnd();
+        }
+        super.visitIdentifier(node);
+    }
+
+    public visitTemplateExpression(node: ts.TemplateExpression) {
```
Comment (ashwinr) [inline src/rules/commentFormatRule.ts]: There's a lot of repetition in these 3 methods. Split out this block into a private addNode method and call it from all these methods

**Pull request merged**